### ISSUE TITLE
Permission denied on Android 13 despite granting

### ISSUE THREAD

**Issue by AySz88** [zamojski/TowerCollector#244]
Workaround: manually download and install the APK via https://github.com/zamojski/TowerCollector/releases

---

I get the permission_collector_never_ask_again_message dialog box whenever I try to start collecting measurements. I tried clearing storage and un-/reinstalling, always granting any permission requests, never checked a "don't ask again" permission request box (never even saw one), and manually granted it every permission Android lists in Settings. I suspect it's related to these messages I notice in logcat:

```
2024-12-04 16:15:38.183 com.google.android.permissioncontroller I Role : info.zamojski.soft.towercollector not qualified for android.app.role.HOME due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.MAIN', mCategories='[android.intent.category.HOME]', mDataScheme='null', mDataType='null'}, mMinTargetSdkVersion=1, mFlags='0', mPermission='null', mQueryFlags=0, mMetaData=[]}
2024-12-04 16:15:38.247 com.google.android.permissioncontroller I Role : info.zamojski.soft.towercollector not qualified for android.app.role.DIALER due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.DIAL', mCategories='[]', mDataScheme='null', mDataType='null'}, mMinTargetSdkVersion=1, mFlags='0', mPermission='null', mQueryFlags=0, mMetaData=[]}
2024-12-04 16:15:38.265 com.google.android.permissioncontroller I Role : info.zamojski.soft.towercollector not qualified for android.app.role.SMS due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.provider.Telephony.SMS_DELIVER', mCategories='[]', mDataScheme='null', mDataType='null'}, mMinTargetSdkVersion=1, mFlags='0', mPermission='android.permission.BROADCAST_SMS', mQueryFlags=0, mMetaData=[]}
```

I'm using 2.15.3 via Google Play.

**Comment by zamojski** [comment]
If you're using Android 13 that has been fixed in 2.15.4. However you'll need to install it from GitHub releases page because Google declined my submission due to some policy violation which was there for 3 years already 🤭 I'm working to fix it but cannot tell when it will be available.

**Comment by zamojski** [comment]
Version 2.15.5 contains the fix and is available in Google Play.
